rust/src/tx_builder.rs

```diff
@@ -367,11 +369,15 @@ impl TransactionBuilder {
                     address: address.clone(),
                     // maximum possible output to maximize fee from adding this output
                     // this may over-estimate the fee by a few bytes but that's okay
-                    amount: to_bignum(0x1_00_00_00_00),
+                    amount: Value::new(to_bignum(0x1_00_00_00_00)),
```

Comment:
Unfortunately this isn't sufficient because the change also needs to contain the difference between assets in the output and the assets in the input (since we need to send any leftover assets back as change)

Reply:
good point. I implemented that, but it doesn't make sense if I don't remove the error that's thrown when computing the min utxo value and there are native assets present.

also, not that it matters for this, but wouldn't `input_total - output_total` be a better upper bound for the fee estimation? After all, the change value would always be lower than that.

Reply:
> wouldn't input_total - output_total be a better upper bound for the fee estimation

I think you're right as long as the `input_total` and `output_total` also take into account deposits, withdrawals, etc. A better estimation would reduce the transaction size by a single-digit number of bytes which is nice but also not super critical either. Up to you if you want to do this, or we can just create an open issue and improve this in the future.

Reply:
I mean, I don't think it's important, I was mostly wondering if I was missing something.
If I implement arithmetic (`checked_add` and `checked_sub`) on `Value` accounting for multiassets too, then I can just use `input_total.checked_sub(output_total)` for the estimation, so I don't mind doing it too.